test/TickMath.starknet.ts

```diff
@@ -0,0 +1,188 @@
+import Decimal from 'decimal.js'
+import {TickMathTest__WC__TickMathTest_compiled} from '../typechain-types'
+import {expect} from 'chai';
+import { getStarknetContractFactory } from 'hardhat-warp/dist/testing'
+import { encodePriceSqrt, MIN_SQRT_RATIO, MAX_SQRT_RATIO } from './shared/utilities'
+import BN from 'bn.js'
+import { BigNumber } from 'ethers'
+
+const MIN_TICK = -887272
+const MAX_TICK = 887272
+
+Decimal.config({ toExpNeg: -500, toExpPos: 500 })
+
+describe('TickMath', () => {
+  let tickMath: TickMathTest__WC__TickMathTest_compiled;
+
+  beforeEach('deploy TickMathTest', async () => {
+    const tickMathFactory = getStarknetContractFactory('TickMathTest')
+    const contract = await tickMathFactory.deploy()
+    tickMath = await (contract.deployed()) as TickMathTest__WC__TickMathTest_compiled
+  })
+
+  describe('#getSqrtRatioAtTick', () => {
+    /*
+    it('throws for too low', async () => {
+      await expect(tickMath.getSqrtRatioAtTick_986cfba3(MIN_TICK - 1)).to.be.revertedWith('T')
+    })
+
+    it('throws for too low', async () => {
+      await expect(tickMath.getSqrtRatioAtTick_986cfba3(MAX_TICK + 1)).to.be.revertedWith('T')
+    })
+    */
+    it('min tick', async () => {
+      const res = await tickMath.getSqrtRatioAtTick_986cfba3(new BN(MIN_TICK).toTwos(24).toString());
+      expect(res[0].toNumber()).to.eq(4295128739)
+    })
+
+    it('min tick +1', async () => {
+      const res = await tickMath.getSqrtRatioAtTick_986cfba3(new BN(MIN_TICK + 1).toTwos(24).toString());
+      expect(res[0].toNumber()).to.eq(4295343490)
+    })
+
+    it('max tick - 1', async () => {
+      const res = await tickMath.getSqrtRatioAtTick_986cfba3(new BN(MAX_TICK - 1).toString());
+      expect(res[0].toString()).to.eq("1461373636630004318706518188784493106690254656249")
+    })
+
+    it('min tick ratio is less than js implementation', async () => {
+      const res = await tickMath.getSqrtRatioAtTick_986cfba3(new BN(MIN_TICK).toTwos(24).toString());
+      const result = encodePriceSqrt(1, BigNumber.from(2).pow(127))
+      expect(res[0].toNumber()).to.be.lt(result.toNumber())
+    })
+
+    it('max tick ratio is greater than js implementation', async () => {
+      const res = await tickMath.getSqrtRatioAtTick_986cfba3(new BN(MAX_TICK).toString());
+      const result = encodePriceSqrt(BigNumber.from(2).pow(127), 1);
+      expect(res[0].toNumber()).to.be.lt(result.toNumber())
```

Comment:
Can you check if this should be `to.be.gt`?

Reply:
Also, since toNumber() is only limited to 53 bits, the following should be a working way to compare the expectations:

```
expect(BigNumber.from(res[0].toString())).to.be.gt(result)
```